source/hic_hal/stm32/stm32h743ii/extended_features/source/DAP_vendor_ex.c

```diff
@@ -283,6 +290,82 @@ uint32_t DAP_ProcessVendorCommandEx(const uint8_t *request, uint8_t *response) {
 
         break;
     }
+    case ID_DAP_VendorEx40_MEASURE_POWER:
+    {
+        bool ret = udb_power_measurement_measure();
+
+        if (ret)
+        {
+            *response = DAP_OK;
+        }
+        else
+        {
+            *response = DAP_ERROR;
+        }
+        num += 1;
+        break;
+    }
+    case ID_DAP_VendorEx41_READ_POWER:
+    {
+        uint8_t target_type = *request++;
+        uint8_t query_type = *request;
+        bool ret;
+        if (target_type < UDB_POWER_MEASUREMENT_TARGET_SIZE)
+        {
+            switch (query_type)
+            {
+                case DAP_VENDOREX_POWER_COMMAND_ARG_VOLTAGE:
+                {
+                    float voltage;
+                    ret = udb_power_measurement_read_voltage_v(target_type, &voltage);
+                    if (ret == false)
+                    {
+                        goto power_measurement_error;
+                    }
+                    *response++ = DAP_OK;
```

Comment:
I wonder if it would be helpful to have some sort of packed struct to show the response format. This setup seems kind of fragile. There could be structs for each command like

    typedef struct __attribute__((packed))
    {
        uint8_t rsp_code;
        uint8_t some_number_i_dont_know;
        int8_t voltage_int; // ?
        uint8_t voltage_frac; // ?
    } dap_vendor_rsp_41_pkt;

or something like that. Then this code would be

    dap_vendor_rsp_41_pkt *pkt = response;
    float voltage_integral;
    float voltage_fractional;
    pkt->rsp_code = DAP_OK;
    pkt->some_number_i_dont_know = 2;
    voltage_fractional = modff(voltage, &voltage_integral); // needs math.h
    pkt->voltage_int = (int8_t) voltage_integral;
    pkt->voltage_frac = (int8_t)(voltage_fractional * 100.0f);
    num += sizeof(dap_vendor_rsp_41_pkt);

which I think is a little more readable and also less error prone in the `num` value. What do you think?

@gaborcsapo any thoughts on that?

Would probably be part of a separate change from this one though.

Reply:
We should clarify who is going to use this power measurement stuff. If it goes to NDM, can we just send the float and let NDM interpret it? @eleenest